### ISSUE TITLE
Blocks doesn't change places after changing hourRowHeight

### ISSUE THREAD

**Issue by iMajeed16** [Skyost/FlutterWeekView#23]
Hello

There is a problem when changing hourRowHeight, the blocks doesn't change to fit the time. So the time starts at 6 AM after changing the hight it still in the old place of the previous hight and that would be now 3 AM, I've tried to rebuild, and building after cleaning but it doesn't work.

**This is before changing**
![This is before changing](https://i.imgur.com/uckWG4C.png)

**This is after changing**
![This is after changing](https://i.imgur.com/hCwL88w.png)

**Smartphone:**
 - Device: S9+
 - OS: Android 10

**Comment by Skyost** [comment]
Can you please provide me your code ?

**Comment by iMajeed16** [comment]
@Skyost

I am using version `flutter_week_view: 0.2.1+7`
and this is my code: 

```dart
DateTime now = DateTime.now();
DateTime date = DateTime(now.year, now.month, now.day);

@override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: StreamBuilder(
            stream: Firestore.instance
                .collection('Task')
                .snapshots(),
            builder: (context, tasks) {
              if (tasks.hasData) {
                List<FlutterWeekViewEvent> events = [];
                for (int i = 0; i < tasks.data.documents.length; i++) {
                  DocumentSnapshot snap = tasks.data.documents[i];

                  events.add(
                    FlutterWeekViewEvent(
                      margin: EdgeInsets.all(0),
                      padding: EdgeInsets.all(0),
                      textStyle: thinWhite,
                      backgroundColor: getColor(
                        snap['state'],
                      ), // return a specific color by state number
                      title: 'Task ${snap['task_id']}',
                      description: 'Task ${snap['task_id']}',
                      start: DateTime.parse(
                        DateFormat('yyyy-MM-dd hh:mm').format(
                          snap['date_time_from'].toDate(),
                        ),
                      ),
                      end: DateTime.parse(
                        DateFormat('yyyy-MM-dd hh:mm').format(
                          snap['date_time_to'].toDate(),
                        ),
                      ),
                      onTap: () => Navigator.push(
                        context,
                        MaterialPageRoute(
                          builder: (context) => ProductsList(
                            documentSnapshot: snap,
                            state: 'details',
                          ),
                        ),
                      ),
                      onLongPress: () {
                        Toast.show('Task ${snap['task_id']}', context,
                            duration: Toast.LENGTH_LONG, gravity: Toast.BOTTOM);
                      },
                    ),
                  );
                }

                return WeekView(
                  hourRowHeight: 100.0,
                  hoursColumnTextStyle: thinWhiteRed,
                  dayBarTextStyle: titleDisable,
                  userZoomable: false,
                  dayViewWidth: MediaQuery.of(context).size.width / 2.5,
                  scrollToCurrentTime: true,
                  dates: [
                    date.subtract(
                      Duration(days: 3),
                    ),
                    date.subtract(
                      Duration(days: 2),
                    ),
                    date.subtract(
                      Duration(days: 1),
                    ),
                    date,
                    date.add(
                      Duration(days: 1),
                    ),
                    date.add(
                      Duration(days: 2),
                    ),
                    date.add(
                      Duration(days: 3),
                    )
                  ],
                  events: events,
                );
              } else {
                return Container();
              }
            }),
      ),
    );
  }

```

Thank you.

**Comment by Skyost** [comment]
Thanks for reporting, I'm gonna test it tomorrow.

**Comment by Skyost** [comment]
Sadly, I can't reproduce this bug. How are you changing `hourRowHeight` ?

**Comment by rosostolato** [comment]
I'm using `flutter_week_view: ^0.3.0` and I had the same behaviour. My `WeekView` is inside a `StreamBuilder` like that example.

Here is my code:

```
  Widget build(BuildContext context) {
    return SafeArea(
      child: Scaffold(
        appBar: AppBar(
          leading: IconButton(
            icon: Icon(Icons.arrow_back),
            onPressed: () => Navigator.pop(context),
          ),
          title: Text('Meus Clientes'),
        ),
        body: StreamBuilder<List<Map<String, dynamic>>>(
          stream: _fetchDocs(),
          builder: (context, snapshot) {
            if (!snapshot.hasData) {
              return Center(child: CircularProgressIndicator());
            }

            return WeekView(
              hourRowHeight: 100.0,
              minimumTime: HourMinute(hour: 7),
              maximumTime: HourMinute(hour: 23),
              dates: List.generate(
                  15, (index) => DateTime.now().add(Duration(days: index))),
              events: snapshot.data.map((doc) {
                Schedule schedule = doc["schedule"];
                Service service = doc["service"];
                var date = schedule.date;

                return FlutterWeekViewEvent(
                  title: schedule.customer.displayName,
                  description: service.description,
                  start: date,
                  end: date.add(Duration(minutes: service.duration)),
                );
              }).toList(),
              style: WeekViewStyle(
                dateFormatter: (year, month, day) => '$day/$month/$year',
                dayViewWidth: MediaQuery.of(context).size.width * 0.6,
              ),
            );
          },
        ),
      ),
    );
  }
```

**Comment by Skyost** [comment]
@rosostolato Please tell me how you are changing `hourRowHeight`. Because in the code of both of you I only see a fixed value.

**Comment by rosostolato** [comment]
I do not change. I set this before launching the app and the display stays like that. And, on the fly, if I remove the property and hot reload, it goes back to default and displays it correctly.

The hour column is changing but the events columns do not. 

Is that something related to the zoom behavior? What do you think?

**Comment by Skyost** [comment]
@rosostolato Well, OP seems to have a problem with changing `hourRowHeight` (see issue title). So you're telling me that this bug occurs while zooming-in and out ?

**Comment by rosostolato** [comment]
@Skyost no sir, forgive me... I was just wondering, I couldn't test so much to try to find out what is happening, I just wanted to collaborate saying that I had that issue too.

And I was saying that I don't need to change the property value on runtime, if the value is not the default one, I will have the same behavior from the screenshots.

**Comment by rosostolato** [comment]
@Skyost by the way, is there a way to start the view zoomed in? Because if there is, I wouldn't need to set RowHeight because it would already solve my problem.

Thank you!

**Comment by Skyost** [comment]
@rosostolato

> And I was saying that I don't need to change the property value on runtime, if the value is not the default one, it will have the same behavior from the screenshots.

Any contribution is welcome :wink:    
So I think I understand now : if you set a value to a non default one (i.e. ≠ 300), then this bug occurs. I thought it occured while changing it dynamically.

> Is there a way to start the view zoomed in ?

Sure, just pinch to zoom in or out (CTRL + drag on emulator).

> You did a good job on it. Thank you!

😉👍

**Comment by rosostolato** [comment]
> No problem, any contribution is welcome !
> So I think I understand now : if you set a value to a non default one (i.e. ≠ 300), then this bug occurs. I thought it occured while changing it dynamically.

Yes, for me the bug happens no matter if you change it dynamically or starts with a fixed value.

> Sure, just pinch to zoom in or out (CTRL + drag on emulator).

But is there a way to start it by default? With full zoom in?

**Comment by Skyost** [comment]
> Yes, for me the bug happens no matter if you change it dynamically or starts with a fixed value.

Okay, thank you. I'm gonna test it this afternoon / evening and will post my feedback here.

> But is there a way to start it by default? With full zoom in?

Oops, I have misread you question. Yes there is a way : just create a custom controller and pass it to your Day View / Week View. Then in your `initState`, wait a frame and call `controller.zoomStart()` and then `controller.zoomUpdate(details)` with your custom [details](https://api.flutter.dev/flutter/gestures/ScaleUpdateDetails-class.html). I will add a `controller.changeZoomFactor(newFactor)` method, but for now you have to do it this way.

**Comment by rosostolato** [comment]
@Skyost cool! Thank you!

**Comment by iMajeed16** [comment]
@Skyost Sorry for being late, thank you @rosostolato for helping with this.

**Comment by iMajeed16** [comment]
```
const ZoomableHeaderWidgetStyle({
    DateFormatter dateFormatter,
    TimeFormatter timeFormatter,
    this.dayBarTextStyle,
    double dayBarHeight,
    this.dayBarBackgroundColor,
    this.hoursColumnTextStyle,
    double hoursColumnWidth,
    this.hoursColumnBackgroundColor,
    double hourRowHeight,
  })  : dateFormatter = dateFormatter ?? DefaultBuilders.defaultDateFormatter,
        timeFormatter = timeFormatter ?? DefaultBuilders.defaultTimeFormatter,
        dayBarHeight = (dayBarHeight ?? 40) < 0 ? 0 : (dayBarHeight ?? 40),
        hoursColumnWidth = (hoursColumnWidth ?? 60) < 0 ? 0 : (hoursColumnWidth ?? 60),
        hourRowHeight = (hourRowHeight ?? 60) < 0 ? 0 : (hourRowHeight ?? 60);
}
```
I did some testing

> hourRowHeight = (hourRowHeight ?? 60) < 0 ? 0 : (hourRowHeight ?? 60);

In this, it takes the 60 no matter what value I gave it won't change after I download the plug-in and added it to my project and change the 60 to the value I gave in 
```
return WeekView(
                  hourRowHeight: 100.0,
 )
```
it works. 

The reason I am changing the height because as you can see the start and end times it is too little and because of that the height of the blocks is too small and it will not show the title nor the description.

![flutter_01](https://user-images.githubusercontent.com/64820019/82218588-c7e88c00-9924-11ea-9f0d-e531b69e2e02.png)


Thanks.

**Comment by Skyost** [comment]
@rosostolato @iMajeed16 Done. I've published `0.3.0+1` on pub.dev, it should be fixed. I've even added a `changeZoomFactor` on controllers. Feel free to try it out !

**Comment by iMajeed16** [comment]
> @rosostolato @iMajeed16 Done. I've published `0.3.0+1` on pub.dev, this issue should be fixed. I've even added a `changeZoomFactor` method on controllers. Feel free to try it out !

everything seems perfect, thank you very much.
